plugin-server/src/utils/db/db.ts

```diff
@@ -997,6 +1002,91 @@ export class DB {
         return kafkaMessages
     }
 
+    public async addPersonOverride(
+        oldPerson: Person,
+        overridePerson: Person,
+        client?: PoolClient
+    ): Promise<ProducerRecord> {
+        const mergedAt = DateTime.now()
+        /** We'll need to do two updates:
+         1. to add an override from oldPerson to override person
+         2. update any entries that have oldPerson as the override person to now also point to the new override person
+
+         TODO: how do we want to be updating oldest_event ? 
+         I'm thinking: we write it if it's not there, but don't update it on conflicts (alternative update to the older date)
+         In the transitive one the same don't update or update to older data (if that's not too complex to do)
+         it's an optimization anyway and squash job might be updating those values anyway and it's just a hint, so ignoring if complex seems better
+         in any case it's important that we use the oldest_event we stored in postgres in CH too.
+        */
+
+        // (1)
+        const {
+            rows: [[version, oldestEvent]],
+        } = await this.postgresQuery(
+            `
+            INSERT INTO posthog_personoverride (
+                team_id, 
+                old_person_id, 
+                override_person_id, 
+                oldest_event,
+                version
+            ) VALUES ($1, $2, $3, $4, $5)
+            ON CONFLICT (team_id, old_person_id) 
+            DO 
```

Comment:
I think we can get rid of this `ON CONFLICT` as it would not be valid to merge `old_person_id` with anything new. Does that make sense @tiina303 ?

Reply:
yup that can be part of the constraint